### PR TITLE
add box center related funcs to box_utils

### LINKED ISSUE #3571: Box utils components
**Is your feature request related to a problem? Please describe.**
The util components used for detection box are not in MONAI

**Describe the solution you'd like**
Add box util components

**Describe alternatives you've considered**
N/A

**Additional context**

### PR DESCRIPTION
**Pull request by Can-Zhao**
Signed-off-by: Can Zhao <canz@nvidia.com>

Fixes #3571.

### Description
add box center related funcs to box_utils

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/data/box_utils.py [hunk 1]
```diff
@@ -572,3 +572,95 @@ def convert_box_to_standard_mode(
...
+    center2 = box_center(boxes2_t.to(compute_dtype))  # (M, spatial_dims)
+
+    if euclidean:
+        dists = (center1[:, None] - center2[None]).pow(2).sum(-1).sqrt()
+    else:
+        # before sum: (N, M, spatial_dims)
+        dists = (center1[:, None] - center2[None]).sum(-1)
+
+    # convert tensor back to numpy if needed
+    (dists, center1, center2), *_ = convert_to_dst_type(src=(dists, center1, center2), dst=boxes1)
```
[the remark this answers is not in the document]
Comment (Can-Zhao) [reply monai/data/box_utils.py:665]: Yes, actually it passed unit test...

[the remark this answers is not in the document]
Comment (Can-Zhao) [reply monai/data/box_utils.py:665]: Please try:
`convert_to_dst_type(src=(np.array([1]),torch.Tensor([2])), dst=torch.Tensor([1]))`

[the remark this answers is not in the document]
Comment (Nic-Ma) [reply monai/data/box_utils.py:665]: OK, I forgot this feature, actually I wrote it before...
https://github.com/Project-MONAI/MONAI/blob/dev/monai/utils/type_conversion.py#L136

Thanks.

**Comment by Can-Zhao** [comment]
> Thanks for the quick PR. Please confirm all the function names, especially "box" vs "boxes", "center" vs "centers", etc. Once we released the APIs, it's hard to change them.
> 
> Thanks.

Thank you! I have updated them

File: docs/source/data.rst [hunk 2]
```diff
@@ -330,3 +330,9 @@ Box mode converter
 .. autofunction:: monai.data.box_utils.get_boxmode
 .. autofunction:: monai.data.box_utils.convert_box_mode
 .. autofunction:: monai.data.box_utils.convert_box_to_standard_mode
+
+Box center
+~~~~~~~~~~
+.. autofunction:: monai.data.box_utils.box_center
```
Comment (Nic-Ma) [inline docs/source/data.rst:336]: function name in the doc should also be updated.

Thanks.

Comment (Can-Zhao) [reply docs/source/data.rst]: Thank you! I have make html and checked it works now.

**Pull request merged**